src/daemon.c

```diff
@@ -293,15 +291,22 @@ int become_daemon(int dont_fork, int close_all_files, const char *user, const ch
 		close(dev_null);
 
 	// generate our pid file
-	{
+	if(pidfile[0]) {
 		unlink(pidfile);
```

Comment:
I am confused again.

There are 2 cases regarding the pidfile:

1. Started are root
2. Started as normal user

Now, we have to decide regarding these:

i. are we going the create a new pidfile or we assume it is already created?
ii. are we going to chown, when we drop privileges?
iii. are we going to truncate it?
iv. are we going to unlink it?

So, can you give me a few guidelines?

PS: `unlink()` removes an entry (unlinks the file) from the directory tree. It has nothing to do with already open files. Open files can be operated as normal, although unlinked.